lib/fastlane/plugin/aws_device_farm/actions/aws_device_farm_action.rb

```diff
@@ -430,6 +430,20 @@ def self.wait_for_run(project, run)
         )
         puts ""
 
+        Dir.mkdir('fastlane/test_output') unless Dir.exist?('fastlane/test_output')
+        Dir.mkdir("fastlane/test_output/#{params[:name]}") unless Dir.exist?("fastlane/test_output/#{params[:name]}")
```

Comment:
checkout https://apidock.com/ruby/FileUtils/mkdir_p - for recursive creation of folders.